modules/sdk-core/src/bitgo/utils/tss/baseTypes.ts

```diff
@@ -0,0 +1,132 @@
+import { SerializedKeyPair } from 'openpgp';
+import { IRequestTracer } from '../../../api';
+import { KeychainsTriplet } from '../../baseCoin';
+import { Keychain } from '../../keychain';
+import { Memo, WalletType } from '../../wallet/iWallet';
+import { EDDSA } from '../../../account-lib/mpc/tss';
+import { EddsaTxRequest } from './eddsa';
+
+export type TxRequestVersion = 'full' | 'lite';
+
+export interface PrebuildTransactionWithIntentOptions {
+  reqId: IRequestTracer;
+  intentType: string;
+  sequenceId?: string;
+  recipients: {
+    address: string;
+    amount: string | number;
+    tokenName?: string;
+  }[];
+  comment?: string;
+  memo?: Memo;
+  tokenName?: string;
+  nonce?: string;
+}
+
+export type TxRequestState =
+  | 'pendingApproval'
+  | 'canceled'
+  | 'rejected'
+  | 'initialized'
+  | 'pendingDelivery'
+  | 'delivered'
+  | 'pendingUserSignature'
+  | 'signed';
+
+export type TransactionState =
+  | 'initialized'
+  | 'pendingSignature'
+  | 'signed'
+  | 'held'
+  | 'delivered'
+  | 'invalidSignature'
+  | 'rejected';
+
+/** Base TxRequest type that different scheme should override  */
+export type ITxRequest = {
+  txRequestId: string;
+  walletId: string;
+  walletType: WalletType;
+  version: number;
+  enterpriseId?: string;
+  state: TxRequestState;
+  date: string;
+  userId: string;
+  intent: unknown; // Should override with sig scheme specific intent
+  pendingApprovalId?: string;
+  policiesChecked: boolean;
+  signatureShares?: SignatureShareRecord[];
+  pendingTxHashes?: string[];
+  txHashes?: string[];
+  // Only available in 'lite' version
+  unsignedTxs: unknown[]; // Should override with blockchain / sig scheme specific unsigned tx
+  // Only available in 'full' version
+  transactions: {
+    state: TransactionState;
+    unsignedTx: unknown; // Should override with blockchain / sig specific unsigned tx
+    signatureShares: SignatureShareRecord[];
+  }[];
+  apiVersion?: TxRequestVersion;
+  latest: boolean;
+};
+
+export enum SignatureShareType {
+  USER = 'user',
+  BACKUP = 'backup',
+  BITGO = 'bitgo',
+}
+
+export interface SignatureShareRecord {
+  from: SignatureShareType;
+  to: SignatureShareType;
+  share: string;
+}
+
+/**
+ * Common Interface for implementing signature scheme specific
+ * util functions
+ */
+export interface ITssUtils<KeyShare = EDDSA.KeyShare, TxRequestType = EddsaTxRequest> {
+  createUserKeychain(
```

Comment:
ecdsa has createParticipantKeychain, can we also add that here or have that in the following pr?